src/globus_sdk/experimental/scope_parser.py

```diff
@@ -4,10 +4,34 @@
   backwards compatibility.
 """
 
-from globus_sdk.scopes import Scope, ScopeCycleError, ScopeParseError
+from __future__ import annotations
+
+import sys
+import typing as t
 
 __all__ = (
     "Scope",
     "ScopeParseError",
     "ScopeCycleError",
 )
+
+if t.TYPE_CHECKING:
+    from globus_sdk.scopes import Scope, ScopeCycleError, ScopeParseError
+else:
+
+    def __getattr__(name: str) -> t.Any:
+        import globus_sdk.scopes as scopes_module
+        from globus_sdk.exc import warn_deprecated
+
+        warn_deprecated(
+            "'globus_sdk.experimental.scope_parser' has been merged into "
+            "'globus_sdk.scopes'. "
+            f"Importing '{name}' from `globus_sdk.experimental` is deprecated. "
+            f"Use `globus_sdk.scopes.{name}` instead."
+        )
```

Comment:
This warning (and possibly this whole module) probably needs to be ripped out before this merges to `4.x-dev`. Thoughts?

Reply:
I tried to be careful about the order here, first to deprecate in v3.x, then to release and merge-back here, and only later (future PR I'm holding) to remove this. Did I botch it? I think this module is still in `4.x-dev`. Applying the change will make the merge history cleaner, I think.

Reply:
If that future PR is coming, great! I'm calling this out because I was not expecting to see a new deprecation warning merge into the 4.x branch, but I suppose this is called the `merge-back` branch, so 🫶 !